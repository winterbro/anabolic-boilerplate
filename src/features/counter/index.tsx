import { Counter, StyledCounter } from 'features/counter/components/Counter';

export { Counter, StyledCounter };
