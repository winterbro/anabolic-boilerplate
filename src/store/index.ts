import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import counterSlice from 'store/slices';

export const store = configureStore({
  reducer: {
    counter: counterSlice.reducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export { counterSlice };
