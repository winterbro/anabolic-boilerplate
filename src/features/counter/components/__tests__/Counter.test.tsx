import { describe, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Counter, StyledCounter } from 'features/counter';
import { Provider } from 'react-redux';
import { store } from 'store';

// Testing page headline
describe('Counter', () => {
  it('should render a counter with the value 0', () => {
    render(
      <Provider store={store}>
        <Counter />)
      </Provider>
    );
    const counter = screen.getByRole('button');
    expect(counter).toHaveTextContent('count is 0');
  });

  it('should increment the counter when clicked', async () => {
    const user = userEvent.setup();
    render(
      <Provider store={store}>
        <Counter />)
      </Provider>
    );
    const counter = screen.getByRole('button');
    await user.click(counter);
    await user.click(counter);
    await user.click(counter);
    expect(counter).toHaveTextContent('count is 3');
  });
});

describe('StyledCounter', () => {
  it('should render the styled component correctly', () => {
    const tree = renderer.create(<StyledCounter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
