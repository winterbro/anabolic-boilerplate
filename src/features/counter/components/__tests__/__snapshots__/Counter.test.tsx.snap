// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StyledCounter > should render the styled component correctly 1`] = `
.c0 {
  padding: 0.6em 1.2em;
  margin-bottom: var(--space-xs);
  font-family: inherit;
  color: #fff;
  cursor: pointer;
  background-color: #1a1a1a;
  border: 1px solid transparent;
  border-radius: 8px;
  -webkit-transition: border - color 0.25s;
  transition: border - color 0.25s;
}

.c0:hover {
  border-color: #646cff;
}

.c0:focus,
.c0:focus-visible {
  outline: 4px auto - webkit - focus - ring - color;
}

<button
  className="c0"
/>
`;
